nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/AbstractCommentManager.java

```diff
@@ -160,4 +162,38 @@ protected void setCommentPermissions(CoreSession session, DocumentModel document
         return ancestorIds;
     }
 
+    /**
+     * @since 11.1
+     */
+    protected DocumentModel getThreadForComment(CoreSession coreSession, DocumentModel comment)
+            throws CommentSecurityException {
+
+        NuxeoPrincipal principal = coreSession.getPrincipal();
+        return CoreInstance.doPrivileged(coreSession, session -> {
+            DocumentModel thread = comment;
+            DocumentModel parent = session.getDocument(new IdRef((String) thread.getPropertyValue(COMMENT_PARENT_ID)));
```

Comment:
This field doesn't exist for previous comment implementation.

Reply:
as we added the new method `org.nuxeo.ecm.platform.comment.api.CommentManager#getAncestorRef` which need the`org.nuxeo.ecm.platform.comment.impl.AbstractCommentManager#getThreadForComment`. and you right this field will not exit in the `CommentManagerImpl`. What we propos is to override `getThreadForComment` in the `CommentManagerIm` which will call its own implementation.

fyi the `CommentManagerImpl` has now its implementation of `getAncestorRef`.

Reply:
Ok, but in the end, what the points to have these two methods in the abstract if it's to override them in one of the two implementations ?

Reply:
yep, we taking into account, the default methods will be removed.